### PR TITLE
[backport] Fix mypy errors

### PR DESCRIPTION
**Pull request by okapies**
Backport of #7423

### CODE REVIEW

**Comment by okapies** [comment]
[automatic post] Jenkins, test this please.

File: chainerx/__init__.pyi [hunk 1]
```diff
@@ -314,6 +314,10 @@ class ndarray:
 
     def __neg__(self) -> ndarray: ...
 
+    def __or__(self, arg0: tp.Any) -> ndarray: ...
+
+    def __pow__(self, arg0: tp.Any) -> ndarray: ...
+
```
Comment (toslunar) [inline chainerx/__init__.pyi]: - ChainerX's `__pow__` is from v7.
- `__or__` looks irrelevant.

Comment (okapies) [reply chainerx/__init__.pyi]: Thanks. It maybe needs to just removes `square` declaration in this backport.

**Comment by toslunar** [comment]
CIs, test this please.
